Mathlib/Analysis/ShiftedLegendrePoly.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2025 Junqi Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junqi Liu
+-/
+import Mathlib.Algebra.Lie.OfAssociative
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Derivative
+import Mathlib.Algebra.Polynomial.Eval.SMul
+import Mathlib.Analysis.RCLike.Basic
+import Mathlib.Data.Nat.Cast.Field
+import Mathlib.Data.Real.StarOrdered
+
+/-!
+# shifted Legendre Polynomials
+
+In this file, we define the shifted Legendre polynomials `shiftedLegendre n` for `n : ℕ` as a
+polynomial in `ℝ[X]`. We prove some basic properties of the Legendre polynomials.
+
+-/
+
+open scoped Nat
+open BigOperators Finset
+
+variable {R : Type*}
+
+namespace Polynomial
+
+/--  `shiftedLegendre n` is the polynomial defined in terms of derivatives of order n.  -/
+noncomputable def shiftedLegendre (n : ℕ) : ℝ[X] :=
+  C (n ! : ℝ)⁻¹ * derivative^[n] (X ^ n * (1 - X) ^ n)
+
+/-- The expand of `shiftedLegendre n`. -/
+theorem shiftedLegendre_eq_sum (n : ℕ) : shiftedLegendre n = ∑ k ∈ Finset.range (n + 1),
+    C ((-1) ^ k : ℝ) • (Nat.choose n k) * (Nat.choose (n + k) n) * X ^ k := by
+  have h : ((X : ℝ[X]) - X ^ 2) ^ n =
+    ∑ m ∈ range (n + 1), n.choose m • (- 1) ^ m * X ^ (n + m) := by
+    rw[sub_eq_add_neg, add_comm, add_pow]
+    congr! 1 with m hm
+    rw[neg_pow, pow_two, mul_pow,← mul_assoc, mul_comm, mul_assoc, pow_mul_pow_sub, mul_assoc,
+      ← pow_add, ← mul_assoc, nsmul_eq_mul, add_comm]
+    rw[Finset.mem_range] at hm
+    linarith
+  rw [shiftedLegendre, ← mul_pow, mul_one_sub, ← pow_two, h, iterate_derivative_sum,
+    Finset.mul_sum]
+  congr! 1 with x _
+  rw [show (n.choose x • (-1) ^ x : ℝ[X]) = C (n.choose x • (-1) ^ x) by simp,
+    Polynomial.iterate_derivative_C_mul, Polynomial.iterate_derivative_X_pow_eq_smul,
+    Nat.descFactorial_eq_div (by omega), show n + x - n = x by omega, smul_eq_mul, nsmul_eq_mul,
+    ← mul_assoc, mul_assoc, mul_comm]
+  simp only [map_mul, map_natCast, map_pow, map_neg, map_one, Algebra.mul_smul_comm,
+    Algebra.smul_mul_assoc, X_pow_mul_assoc_C]
+  rw [Algebra.smul_def, algebraMap_eq, map_natCast, add_comm, Nat.add_choose, ← mul_assoc,
+    ← mul_assoc]
+  congr 1
+  rw [mul_rotate, mul_assoc]
+  nth_rewrite 2 [mul_comm]
```

Comment:
Instead of `nth_rewrite` can you use `mul_comm foo bar` inside the previous `rw`?